### DISCUSSION TITLE
Using slint as a wrapper around a game engine like Bevy

### DISCUSSION THREAD

**Discussion by jrmoulton** [slint-ui/slint#940] [Q&A] [answered]
Is it possible to use slint as a wrapper for UI elements around or also inside of a scene rendered by a game engine like Bevy?

**Comment by hunger** [comment]
I should be possible to use slint in an uderlay/overlay to the scene rendered by a game engine at this time pretty easily.

Mixing both (like having slint render into a virtual screen shown by the game engine or embedding a game engine scene into the slint window is probably harder.

So kt kind of depends on what you want to do:-)

**Comment by tronical** [comment, marked as answer]
Bevy is using wgpu to render its scene, which *can* be done using OpenGL ES 3.0. So in theory it may seem possible to combine them. In practice this isn't going to work out of the box, I think it'll require changes on the bevy side as well as Slint.

The basic idea would be to render bevy as an underlay, by bevy and Slint sharing the same command encoder (create at least two render passes). There's a lot of glue missing for that. I hope in the future Slint can support wgpu and such a scenario becomes feasible.

**Comment by tronical** [comment]
Update: We're working on an example that shows how to integrate Bevy into Slint: https://github.com/slint-ui/slint/pull/8380
